src/common/thread_measurement.c

```diff
@@ -3,14 +3,31 @@
 #include "cmsis_os.h" //osDelay
 #include "filament_sensor.h"
 #include "marlin_client.h"
+#include "trinamic.h"
 
 void StartMeasurementTask(void const *argument) {
     marlin_client_init();
+
+    // wait for startup
+    while (!marlin_event_clr(MARLIN_EVT_StartProcessing)) {
```

Comment:
This is a common function, which may be used by other tasks too. I recommend extracting the implementation into e.g.
`marlin_client_wait_for_start_processing()` or similar.

Reply:
marlin_event_clr is comon function for read and clear event state in single call. We do not need create own function for each event. This is special case of usage because we are waiting in loop (we need block until server ready, but also we need process client loop). I can allocate signal or mutex and synchronize startup using RTOS, but this is more simple and also more readable (principle is not hidden). My opinion is that using osDelay for synchronisation is proper way if we do not need react immediately (for example in DMA synchronization we are using signals).

Reply:
All what you wrote is acceptable, but it doesn't answer the first part of the comment - create a function `marlin_client_wait_for_start_processing()` which **does hide the implementation on purpose** and which also can be improved in the future by using OS synchronization primitives at one place.
I'd really like to avoid having each task implementing its own startup waiting (which is what you propose), when we want all tasks to wait the same way.

Reply:
commit faa7a74